### PR TITLE
Only allow function-like listeners

### PR DESCRIPTION
**Pull request by kylewiedman**
when emitted, Non-functions throw when apply is attempted.

### CODE REVIEW

File: EventEmitter.js [hunk 1]
```diff
@@ -54,6 +54,17 @@
...
     /**
+     * Check to see if an object is a function.
+     *
+     * @param {Object} the object to check.
+     * @return {Bool} Is a function or not
+     * @api private
+    */
+    function isFunction(x) {
+      return Object.prototype.toString.call(x) == '[object Function]';
+    }
```
Comment (Olical) [inline EventEmitter.js]: Why not `typeof x === 'function'`?

File: EventEmitter.js [hunk 2]
```diff
@@ -130,10 +141,17 @@
      * @return {Object} Current instance of EventEmitter for chaining.
      */
     proto.addListener = function addListener(evt, listener) {
+
+        if (listener === null) { return this; }
```
Comment (Olical) [inline EventEmitter.js]: If you're checking for `null` + `undefined` this should be `==`, not `===`. Not sure I like the silent failure either. I'd rather it let you add null or throw a type error.

Just checked: Node throws a type error.

> TypeError: listener must be a function

File: EventEmitter.js [hunk 3]
```diff
@@ -130,10 +141,17 @@
...
+
         var listeners = this.getListenersAsObject(evt);
         var listenerIsWrapped = typeof listener === 'object';
         var key;
 
+
+        if (!listenerIsWrapped && (!isFunction(listener))) { return this; }
+        else if (listenerIsWrapped && (!isFunction(listener.listener))) { return this; }
+
```
Comment (Olical) [inline EventEmitter.js]: This block is pretty confusing to me. I think this PR should just be:

```js
if (typeof listener !== 'function') {
    throw new TypeError('listener must be a function');
}
```

**Comment by Olical** [comment]
My approach is here https://github.com/Olical/EventEmitter/pull/113/commits/64f2a38c0da8f60506554cbcf848164b9bcf3ccb

**Comment by kylewiedman** [comment]
I agree your approach is better and cleaner

**Pull request closed without merging**